bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/debug/DebugModeHandler.java

```diff
@@ -205,15 +220,108 @@ public String calculateDebugPort(Project project, String inputParms) throws Exce
 
     }
 
+    public static AttachingConnector getAttachingConnector() {
+        List<?> connectors = Bootstrap.virtualMachineManager().attachingConnectors();
+        for (int i = 0; i < connectors.size(); i++) {
+            AttachingConnector c = (AttachingConnector) connectors.get(i);
+            if ("com.sun.jdi.SocketAttach".equals(c.name()))
+                return c;
+        }
+
+        return null;
+    }
+
+    /**
+     * Configure the connector properties.
+     *
+     * @param map argument map
+     * @param host the host name or IP address
+     * @param portNumber the port number
+     */
+    public static void configureConnector(Map map, String host, int portNumber) {
```

Comment:
If methods are only called by the class maybe they should be private/protected for clarity?